blockchain/txSigner.go

```diff
@@ -1,41 +1,61 @@
 package blockchain
 
 import (
+	"github.com/ElrondNetwork/elrond-go-core/hashing/keccak"
 	"github.com/ElrondNetwork/elrond-go-crypto"
 	"github.com/ElrondNetwork/elrond-go-crypto/signing"
 	"github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519"
 	"github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519/singlesig"
 )
 
-var suite = ed25519.NewEd25519()
-var singleSigner = &singlesig.Ed25519Signer{}
+var (
+	suite         = ed25519.NewEd25519()
+	hasher        = keccak.NewKeccak()
+	singleSigner  = &singlesig.Ed25519Signer{}
+	messagePrefix = []byte("\x17Elrond Signed Message:\n")
+)
 
-// txSigner contains the primitives used to correctly sign a transaction
-type txSigner struct {
+// xSigner contains the primitives used to correctly sign a transaction
+type xSigner struct {
 	keyGen crypto.KeyGenerator
 }
 
-// NewTxSigner will create a new instance of txSigner
-func NewTxSigner() *txSigner {
-	return &txSigner{
+// NewXSigner will create a new instance of xSigner
+func NewXSigner() *xSigner {
+	return &xSigner{
 		keyGen: signing.NewKeyGenerator(suite),
 	}
 }
 
-// SignMessage will generate the signature providing the private key bytes and the message bytes (usually the serialized
-// form of the transaction)
-func (ts *txSigner) SignMessage(msg []byte, skBytes []byte) ([]byte, error) {
-	sk, err := ts.keyGen.PrivateKeyFromByteArray(skBytes)
+// SignMessage will generate the signature providing the private key bytes and the message bytes
```

Comment:
L31-L54 can be refactored as follows:
```GO
// SignMessage will generate the signature providing the private key bytes and the message bytes
// prepending the standard messagePrefix
func (xs *xSigner) SignMessage(msg []byte, skBytes []byte) ([]byte, error) {
	serializedMessage, err := xs.serializeForSigning(msg)
	if err != nil {
		return nil, err
	}

	return xs.SignByteSlice(serializedMessage, skBytes)
}

// SignByteSlice will generate the signature providing the private key bytes and the byte slice to be signed
func (xs *xSigner) SignByteSlice(buff []byte, skBytes []byte) ([]byte, error) {
	sk, err := xs.keyGen.PrivateKeyFromByteArray(skBytes)
	if err != nil {
		return nil, err
	}

	return singleSigner.Sign(sk, buff)
}
```
you will improve code reuse and rename the somehow misleading `SignTransaction` function which is not related to a transaction at all.

Reply:
refactored